src/digest/digest.rs

```diff
@@ -296,6 +306,18 @@ pub struct Algorithm {
     pub nid: c::int,
 }
 
+impl fmt::Debug for Algorithm {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self.nid {
+            64 => fmt.write_str("SHA-1"),
+            672 => fmt.write_str("SHA-256"),
+            673 => fmt.write_str("SHA-384"),
+            674 => fmt.write_str("SHA-512"),
+            _ => unreachable!(),
```

Comment:
I didn't find public constructors for `Algorithm` therefore I concluded panicking would be ok.

Alhough I think that `core::fmt::Error` has the semantics that the process of formatting failed and not that the field has an unexpected value I can make this change. What about just outputting "NID {self.nid}"?